src/truffle-integration/projectFsWatcher.js

```diff
@@ -89,18 +99,19 @@ class ProjectFsWatcher extends EventEmitter {
           ) {
             this.startWatchingContracts();
           }
+
+          if (tail[0] && filename === path.basename(tail[0]))
```

Comment:
Perhaps this and lines 97/98 (won't let met comment there) should not be using the basename function, and be more something like

`path.join(head, filename) === path.normalize(this.project.config.contracts_build_directory)`

I don't think it's an issue either way as `startWatchingContracts()` will check to see if the directory exists before watching. Food for thought